app/Http/Controllers/App/PostController.php

```diff
@@ -59,11 +59,59 @@ public function store(Request $request)
         return redirect()->route('app.post.index');
     }
 
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Inertia\Response
+     */
+    public function edit(Request $request, int $id)
+    {
+        $post = Post::with('team.challenges', 'team.currentChallenges', 'team.banner')->findOrFail($id);
+
+        abort_unless($post->canEdit($request), 403, 'Access denied. Only the Author can edit the post');
```

Comment:
This is a perfect usecase for a `Policy`: https://laravel.com/docs/8.x/authorization#creating-policies

```suggestion
        abort_unless($request->user()->can('edit', $post), 403, 'Access denied. Only the Author can edit the post');
```